toolkits/math/arcade_math/tools/arithmetic.py

```diff
@@ -1,75 +1,268 @@
 
 
 @tool
 def divide(
-    a: Annotated[int, "The first number"], b: Annotated[int, "The second number"]
-) -> Annotated[float, "The quotient of the two numbers"]:
+    a: Annotated[str, "The first number as a string"],
+    b: Annotated[str, "The second number as a string"],
+) -> Annotated[str, "The quotient of the two numbers as a string"]:
     """
-    Divide two numbers
+    Divide two numbers.
     """
-    return a / b
+    # Use Decimal for arbitrary precision
+    a_decimal = Decimal(a)
+    b_decimal = Decimal(b)
+    return str(a_decimal / b_decimal)
 
 
 @tool
 def sqrt(
-    a: Annotated[int, "The number to square root"],
-) -> Annotated[float, "The square root of the number"]:
+    a: Annotated[str, "The number to square root as a string"],
+) -> Annotated[str, "The square root of the number as a string"]:
     """
     Get the square root of a number
+    If 
     """
-    return math.sqrt(a)
+    # Use Decimal for arbitrary precision
+    a_decimal = Decimal(a)
+    return str(a_decimal.sqrt())
 
 
 @tool
 def sum_list(
-    numbers: Annotated[list[float], "The list of numbers"],
-) -> Annotated[float, "The sum of the numbers in the list"]:
+    numbers: Annotated[list[str], "The list of numbers as strings"],
+) -> Annotated[str, "The sum of the numbers in the list as a string"]:
     """
     Sum all numbers in a list
     """
-    return sum(numbers)
+    # Use Decimal for arbitrary precision
+    return str(sum([Decimal(n) for n in numbers]))
 
 
 @tool
 def sum_range(
-    start: Annotated[int, "The start of the range to sum"],
-    end: Annotated[int, "The end of the range to sum"],
-) -> Annotated[int, "The sum of the numbers in the list"]:
+    start: Annotated[str, "The start of the range to sum as a string"],
+    end: Annotated[str, "The end of the range to sum as a string"],
+) -> Annotated[str, "The sum of the numbers in the list as a string"]:
     """
     Sum all numbers from start through end
     """
-    return sum(list(range(start, end + 1)))
+    return str(sum(list(range(int(start), int(end) + 1))))
+
+
+# -----------------------
+# Additional useful tools
+# -----------------------
+
+
+@tool
+def mod(
+    a: Annotated[str, "The dividend as a string"],
+    b: Annotated[str, "The divisor as a string"],
+) -> Annotated[str, "The remainder after dividing a by b as a string"]:
+    """
+    Calculate the remainder (modulus) of one number divided by another.
+    """
+    # Use Decimal for arbitrary precision
+    return str(Decimal(a) % Decimal(b))
+
+
+@tool
+def power(
+    a: Annotated[str, "The base number as a string"],
+    b: Annotated[str, "The exponent as a string"],
+) -> Annotated[str, "The result of raising a to the power of b as a string"]:
+    """
+    Calculate one number raised to the power of another.
+    """
+    # Use Decimal for arbitrary precision
+    return str(Decimal(a)**Decimal(b))
+
+
+@tool
+def abs_val(
+    a: Annotated[int, "The number as a string"],
```

Comment:
int or str?

Reply:
it should be str, fixed now, together with linter checks